packages/broker/src/plugins/operator/MaintainTopologyService.ts

```diff
@@ -1,68 +1,110 @@
 
 const logger = new Logger(module)
 
+/**
+ * Helper function for setting up and starting a MaintainTopologyService along
+ * with all its dependencies.
+ */
+export async function setUpAndStartMaintainTopologyService({
+    streamrClient,
+    replicationFactor,
+    serviceHelperConfig,
+    operatorFleetState
+}: {
+    streamrClient: StreamrClient
+    replicationFactor: number
+    serviceHelperConfig: OperatorServiceConfig
+    operatorFleetState?: OperatorFleetState
+}): Promise<MaintainTopologyService> {
+    const operatorFleetStateGiven = operatorFleetState !== undefined
+    if (operatorFleetState === undefined) {
+        const coordinationStreamId = toStreamID('/operator/coordination', serviceHelperConfig.operatorContractAddress)
+        operatorFleetState = new OperatorFleetState(streamrClient, coordinationStreamId)
+    }
+    const maintainTopologyHelper = new MaintainTopologyHelper(serviceHelperConfig)
+    const nodeId = (await streamrClient.getNode()).getNodeId()
+    const service = new MaintainTopologyService(
+        streamrClient,
+        new StreamAssignmentLoadBalancer(
+            nodeId,
+            replicationFactor,
+            async (streamId) => {
+                const stream = await streamrClient.getStream(streamId)
+                return stream.getStreamParts()
+            },
+            operatorFleetState,
+            maintainTopologyHelper
+        )
+    )
+    await service.start()
+    if (!operatorFleetStateGiven) {
+        await operatorFleetState.start()
+    }
+    await maintainTopologyHelper.start()
+    return service
+}
+
 export class MaintainTopologyService {
     private readonly streamrClient: StreamrClient
-    private readonly maintainTopologyHelper: MaintainTopologyHelper
-    private readonly subscriptions = new Multimap<StreamID, Subscription>()
+    private readonly streamAssignmentLoadBalancer: StreamAssignmentLoadBalancer
+    private readonly subscriptions = new Map<StreamPartID, Subscription>()
     private readonly concurrencyLimit = pLimit(1)
 
-    constructor(streamrClient: StreamrClient, maintainTopologyHelper: MaintainTopologyHelper) {
+    constructor(streamrClient: StreamrClient, streamAssignmentLoadBalancer: StreamAssignmentLoadBalancer) {
         this.streamrClient = streamrClient
-        this.maintainTopologyHelper = maintainTopologyHelper
+        this.streamAssignmentLoadBalancer = streamAssignmentLoadBalancer
     }
 
     async start(): Promise<void> {
-        this.maintainTopologyHelper.on('addStakedStream', this.onAddStakedStreams)
-        this.maintainTopologyHelper.on('removeStakedStream', this.onRemoveStakedStream)
-        await this.maintainTopologyHelper.start()
+        this.streamAssignmentLoadBalancer.on('assigned', this.onAddStakedStreamPart)
+        this.streamAssignmentLoadBalancer.on('unassigned', this.onRemoveStakedStreamPart)
         logger.info('Started')
     }
 
+    // eslint-disable-next-line class-methods-use-this
```

Comment:
Remove the method as it doesn't do anything?

Reply:
Removed in https://github.com/streamr-dev/network/pull/1643/commits/af15d795c74b3596be525459ca469f501ae2c137